combat-trainer.lic

```diff
@@ -1910,14 +1919,38 @@ class GameState
     @use_weak_attacks = false
   end
 
+  def use_analyze_flame?
+    @use_analyze_flame
+  end
+
+  def stop_analyze_flame
+    @use_analyze_flame = false
+  end
+
   def melee_attack_verb
```

Comment:
I'm worried about an edge case that I've encountered before.

I do not see anything in the "switch to a new skill" logic that prevents you from changing skills mid-combo. If the before and after skills are not compatible (i.e. use different moves), it's possible to still have a combo going which you cannot complete with your current weapon. For example, try `combat-trainer` using only Brawling and an Edged weapon.

Did you test this case?